ValveResourceFormat/Resource/ResourceTypes/Choreo/Parser/BVCDParser.cs

```diff
@@ -0,0 +1,357 @@
+            for (var i = 0; i < count; i++)
+            {
+                relativeTags[i] = ReadRelativeTag();
+            }
+
+            //flex timing tags
+            count = reader.ReadByte();
+            var flexTimingTags = new ChoreoFlexTimingTag[count];
+            for (var i = 0; i < count; i++)
+            {
+                flexTimingTags[i] = ReadFlexTimingTag();
+            }
+
+
+            //absolute tags
+            //play tags
+            count = reader.ReadByte();
+            var playTags = new ChoreoAbsoluteTag[count];
+            for (var i = 0; i < count; i++)
+            {
+                playTags[i] = ReadAbsoluteTag();
+            }
+
+            //shift tags
+            count = reader.ReadByte();
+            var shiftTags = new ChoreoAbsoluteTag[count];
+            for (var i = 0; i < count; i++)
+            {
+                shiftTags[i] = ReadAbsoluteTag();
+            }
+
+            var sequenceDuration = -1f;
+            if (eventType == ChoreoEventType.Gesture)
+            {
+                sequenceDuration = reader.ReadSingle();
+            }
+
+            var usingRelativeTag = reader.ReadBoolean();
+            ChoreoRelativeTag relativeTag = null;
+            if (usingRelativeTag)
+            {
+                relativeTag = ReadRelativeTag();
+            }
+
+            var flex = ReadFlex();
+
+            byte loopCount = 0;
+            ChoreoClosedCaptions closedCaptions = null;
+            if (eventType == ChoreoEventType.Loop)
+            {
+                loopCount = reader.ReadByte();
+            }
+            else if (eventType == ChoreoEventType.Speak)
+            {
+                if (version < 17)
+                {
+                    closedCaptions = ReadClosedCaptions();
+                }
+                var unk02 = reader.ReadInt32(); //is this for speak only? what's this? maybe part of closedcaptions?
+                Debug.Assert(unk02 == 0);
+            }
+
+            if (version >= 17)
+            {
+                var unk03 = reader.ReadByte();
+            }
+
+            //eventId or unk01 is sometimes missing?
+            var unk01 = reader.ReadInt32(); //this is likely a reference to another event's id, act1_refuge.vcdlist has non-zero values here
+            Debug.Assert(unk01 == 0);
+            var eventId = reader.ReadInt32();
+
+            var absoluteTags = playTags.Concat(shiftTags).ToArray();
+            return new ChoreoEvent(eventType, name, eventStart, eventEnd, param1, param2, param3, ramp, flags, distanceToTarget, relativeTags, flexTimingTags, absoluteTags, sequenceDuration, usingRelativeTag, relativeTag, flex, loopCount, closedCaptions, eventId, unk01);
+        }
+        protected virtual ChoreoClosedCaptions ReadClosedCaptions()
+        {
+            var type = (ChoreoClosedCaptionsType)reader.ReadByte();
+            var token = ReadString();
+            var flags = (ChoreoClosedCaptionsFlags)reader.ReadByte();
```

Comment:
There's also SoundStartDelay float after these flags.

There's also flag 8, but i'm unsure what it means.